src/libpgmoneta/wal.c

```diff
@@ -211,28 +213,41 @@ pgmoneta_wal(int srv, char** argv)
    }
    config->servers[srv].cur_timeline = cur_timeline;
 
-   wal_find_streaming_start(d, &timeline, &high32, &low32, segsize);
-   if (timeline == 0)
-   {
-      // use current xlogpos as last resort
-      timeline = cur_timeline;
-      xlogpos_size = strlen(pgmoneta_query_response_get_data(identify_system_response, 2)) + 1;
-      xlogpos = (char*)malloc(xlogpos_size);
+   read_replication = (config->servers[srv].version >= 15) ? 1 : 0;
 
-      if (xlogpos == NULL)
+   if (read_replication)
```

Comment:
Remember this is only the last resort. Normally we get our starting point by finding the latest wal segment in the wal archive directory, say `backup/primary/wal`. So you should read replication slot only if that archive directory is empty, meaning `wal_find_streaming_start` gets a 0 timeline.